### ISSUE TITLE
VAE reconstruction loss 

### ISSUE THREAD

**Issue by mandrecki** [AppliedDataSciencePartners/WorldModels#2]
Wow! Cool to see a reproduction so fast! Thanks a lot!

Shouldn't the reconstruction loss:

`        def vae_r_loss(y_true, y_pred):

            return K.sum(K.square(y_true - y_pred), axis = [1,2,3])
`
be based on the mean rather than sum?

If it's sum-based than its value depends on the size of the image. Additionally it throws off the balance between KL and reconstruction loss -- the larger the image, the more reconstruction loss overwhelms the optimisation.

**Comment by Bleyddyn** [comment]
Could this be related to the NaN's I'm getting when trying to run the vae training step?

First the KL loss (and overall loss) blows up, then starts going back down, but turns into NaN's before long:

13696/48000 [=======>......................] - ETA: 4:32 - loss: 289661273.9050 - vae_r_loss: 0.1846 - vae_kl_loss: 289661273.7244
23904/48000 [=============>................] - ETA: 3:13 - loss: 191863659.1293 - vae_r_loss: 0.1638 - vae_kl_loss: 191863658.9721
24992/48000 [==============>...............] - ETA: 3:04 - loss: nan - vae_r_loss: nan - vae_kl_loss: nan                          


Mac OS X 10.13
Python 3.6.2
tf.__version__ '1.7.0'
keras.__version__ '2.1.5'
gym.__version__ '0.10.5'
